src/shared/Atlassian.Bitbucket/BitbucketHostProvider.cs

```diff
@@ -83,16 +92,56 @@ public async Task<ICredential> GetCredentialAsync(InputArguments input)
             {
                 throw new Exception("Unencrypted HTTP is not supported for Bitbucket.org. Ensure the repository remote URL is using HTTPS.");
             }
+        }
+
+        private async Task<ICredential> GetStoredCredentials(InputArguments input)
+        {
+            if(_context.Settings.TryGetSetting(BitbucketConstants.EnvironmentVariables.AlwaysRefreshCredentials, 
+                Constants.GitConfiguration.Credential.SectionName, BitbucketConstants.GitConfiguration.Credential.AlwaysRefreshCredentials, 
+                out string alwaysRefreshCredentials) && alwaysRefreshCredentials.ToBooleanyOrDefault(false))
+            {
+                _context.Trace.WriteLine($"Ignore stored credentials");
+                return null;
+            }
+
+            var targetUri = input.GetRemoteUri();
+            string credentialService = GetServiceName(input);
+            _context.Trace.WriteLine($"Look for existing credentials under {credentialService} ...");
+
+            var credentials = _context.CredentialStore.Get(credentialService, input.UserName);
+
+            if(credentials == null)
+            {
+                _context.Trace.WriteLine($"    Found none");
+                return null;
+            }
+
+            _context.Trace.WriteLine($"    Found credentials: {credentials.Account}/**********");
```

Comment:
If you wanted you could add the `credentials.Password` field here, and use `Trace.WriteLineSecrets` appropriately to mask the password/token.